### PR TITLE
adding hostNetwork and dnsPolicy to helm chart

### PR DESCRIPTION
**Pull request by real-danm**
This PR allows you to configure hostNetwork'ing mode along with a dnsPolicy on the nats statefulset. This can be useful when nats traffic is required to run on specific interfaces attached to a VM.

### CODE REVIEW

File: helm/charts/nats/values.yaml [hunk 1]
```diff
@@ -329,6 +329,13 @@ affinity: {}
 ## ref: https://kubernetes.io/docs/concepts/configuration/pod-priority-preemption/#priorityclass
 priorityClassName: null
 
+## hostNetwork
+hostNetwork: false
```
Comment (caleblloyd) [inline helm/charts/nats/values.yaml:333]: should be under `nats` key as this only applies to the stateful set

File: helm/charts/nats/values.yaml [hunk 2]
```diff
@@ -329,6 +329,13 @@ affinity: {}
...
+
+## Pod Dns Policy. Default is ClusterFirst
+## ref: https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy
+dnsPolicy: ClusterFirst
```
Comment (caleblloyd) [inline helm/charts/nats/values.yaml:337]: should be under `nats` key as this only applies to the stateful set

File: helm/charts/nats/templates/statefulset.yaml [hunk 3]
```diff
@@ -81,6 +81,12 @@ spec:
       {{- if .Values.priorityClassName }}
       priorityClassName: {{ .Values.priorityClassName | quote }}
       {{- end }}
+      {{- with .Values.dnsPolicy }}
+      dnsPolicy: {{ toYaml . | nindent 8 }}
```
Comment (caleblloyd) [inline helm/charts/nats/templates/statefulset.yaml:85]: this is always a string, so just put the string here, no need for `toYaml`

File: helm/charts/nats/templates/statefulset.yaml [hunk 4]
```diff
@@ -81,6 +81,12 @@ spec:
...
+      {{- end }}
+      {{- with .Values.hostNetwork }}
+      hostNetwork: {{ toYaml . | nindent 8 }}
```
Comment (caleblloyd) [inline helm/charts/nats/templates/statefulset.yaml:88]: this is always a bool, so just put the bool here, no need for `toYaml`

**Review by caleblloyd — CHANGES_REQUESTED** [review]
Thanks for the contribution.  Please add a description and briefly describe what this PR does.

**Review by caleblloyd — APPROVED** [review]

**Comment by caleblloyd** [comment]
Looks good to me, @wallyqs could you also take a look?

**Review by wallyqs — APPROVED** [review]
LGTM

**Pull request merged**